### PR TITLE
fix(core) add a replacer to handle functions/classes when creating mo…

### PR DESCRIPTION
**Pull request by zenozen**
…dule tokens

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #2379

## What is the new behavior?
ModuleTokenFactory now correctly handles metadata with class/function objects in.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: packages/core/injector/module-token-factory.ts [hunk 1]
```diff
@@ -50,4 +50,12 @@ export class ModuleTokenFactory {
     const scope = Reflect.getMetadata(SHARED_MODULE_METADATA, metatype);
     return scope ? scope : 'global';
   }
+
```
Comment (BrunnerLivio) [inline packages/core/injector/module-token-factory.ts]: Can you add a comment about why this replacer function exists in the first place? So future contributors won't get confused and delete this function :)

**Comment by zenozen** [comment]
@BrunnerLivio Ah, will do. I tried to run all integration test suite and failed with some syntax error. Your command does successfully run the injector integration tests. Will make the change tonight after getting off work, thanks for the review :)

File: packages/core/injector/module-token-factory.ts [hunk 2]
```diff
@@ -50,4 +50,12 @@ export class ModuleTokenFactory {
...
+  private replacer(key: string, value: any) {
+    if (typeof value === 'function') {
+      return value.name;
+    } else {
+      return value;
+    }
```
Comment (rhogeranacleto) [inline packages/core/injector/module-token-factory.ts]: :bulb: tip: You dont need this `else` statement.
```ts
    if (typeof value === 'function') {
      return value.name;
    }
    return value;
```

**Comment by kamilmysliwiec** [comment]
I have added a comment :) Amazing contribution, thank you!

**Pull request merged**